ci-operator/step-registry/azure/cluster/custom-dns/azure-cluster-custom-dns-ref.metadata.json

```diff
@@ -0,0 +1,12 @@
+{
+	"path": "azure/cluster/custom-dns/azure-cluster-custom-dns-ref.yaml",
```

Comment:
How about keep consistent naming like gcp and aws as `azure-provision-custom-dns` ?